examples/template_alignment.py

```diff
@@ -0,0 +1,164 @@
+template_estim.fit(template_train, n_iter=2)
+
+#############################################################################
+# Predict subject 7 PA data from the template and fitted estimator
+# ----------------------------------------------------------------
+# We use target_train data to fit the transform, indicating it corresponds to
+# the contrasts indexed by train_index and predict from this learnt alignment
+# contrasts corresponding to template test_index numbers.
+
+# For each train subject and for the template, the AP contrasts are sorted from
+# 0, to 53, and then the PA contrasts from 53 to 106.
+train_index = range(53)
+test_index = range(53, 106)
+# We input the mapping image target_train in a list, we could have input more
+# than one subject for which we'd want to predict : [train_1, train_2 ...]
+prediction_from_template = template_estim.transform([target_train], train_index,
+                                                    test_index)
+# We can also try to predict from averaging
+prediction_from_average = index_img(average_subject, test_index)
+
+#############################################################################
+# Score the prediction of test data without alignment
+# ---------------------------------------------------
+# To score the quality of prediction we use r2 score on each voxel \
+# activation profile across contrasts. This score is 1 for a perfect prediction \
+# and can get arbitrarly bad (here we clip it to -1 for bad predictions)
+# We compare accuracy of predictions made from group average and from template.
+
+import numpy as np
+from sklearn.metrics import r2_score
+# The PA contrasts deduced from averaging training subjects
+
+# The baseline score represents the quality of prediction using raw data
+average_score = np.maximum(r2_score(
+    masker.transform(target_test), masker.transform(prediction_from_average),
+    multioutput='raw_values'), -1)
+# The baseline score represents the quality of prediction using aligned data
+template_score = np.maximum(r2_score(
+    masker.transform(target_test), masker.transform(
+        prediction_from_template[0]),
+    multioutput='raw_values'), - 1)
+
+#############################################################################
+# Plotting the prediction quality
+# ---------------------------------------------------
+#
+from nilearn import plotting
+
+baseline_display = plotting.plot_stat_map(masker.inverse_transform(
+    average_score), display_mode="z", vmax=1, cut_coords=[-15, -5])
+baseline_display.title("R2 score of prediction from group average")
+
+display = plotting.plot_stat_map(
+    masker.inverse_transform(
+        template_score), display_mode="z", cut_coords=[-15, -5], vmax=1)
+display.title("R2 score of prediction using template and alignment")
+plotting.show()
+#############################################################################
+# We observe that creating a template and aligning a new subject to it enables
```

Comment:
aligning... to it yields  better prediction of his  test contrasts than just using the group average.